### PR TITLE
new flo2D mesh creation approach

### PR DESCRIPTION
**Pull request by vcloarec**
fix FLO-2D 2D mesh loading for big space cover and improve loading

### CODE REVIEW

File: mdal/frmts/mdal_flo2d.cpp [hunk 1]
```diff
@@ -494,21 +490,41 @@ void MDAL::DriverFlo2D::parseFPLAINFile( std::vector<double> &elevations,
...
+    if ( !cellSizeCalculated )
     {
-      cells[cc_i].conn[j] = MDAL::toInt( lineParts[j + 1] ) - 1; //numbered from 1, 0 boundary Vertex
+      size_t cc_i = MDAL::toSizeT( lineParts[0] ) - 1; //numbered from 1
+      for ( int i = 1; i < 5; ++i )  //search the first cell that have a neighbor to calculate cell size
+      {
+        int neighborCell = MDAL::toInt( lineParts[i] );
+        if ( neighborCell != 0 )
+        {
+          if ( i % 2 == 1 ) //North or South neighbor
```
Comment (PeterPetrik) [inline mdal/frmts/mdal_flo2d.cpp:512]: typo: neighbor

File: mdal/frmts/mdal_flo2d.cpp [hunk 2]
```diff
@@ -494,21 +490,41 @@ void MDAL::DriverFlo2D::parseFPLAINFile( std::vector<double> &elevations,
...
+            cellSize = fabs( cells[neighborCell - 1].y - cells[cc_i].y );
+          else // east or West
```
Comment (PeterPetrik) [inline mdal/frmts/mdal_flo2d.cpp:514]: East or West

**Review by PeterPetrik — COMMENTED** [review]
Looks like the FLO-2D link is missing here: https://github.com/lutraconsulting/MDAL/blob/master/docs/source/drivers/index.rst

Also if you can please look at https://github.com/lutraconsulting/MDAL/blob/master/docs/source/drivers/flo2d.rst and see if it is OK or there is something you can add now when you have all the info in memory...

File: mdal/frmts/mdal_flo2d.cpp (hunk 1 above)
Comment (vcloarec) [reply mdal/frmts/mdal_flo2d.cpp:512]: ?

**Comment by vcloarec** [comment]
> see if it is OK or there is something you can add now when you have all the info in memory...

nothing to add

File: mdal/frmts/mdal_flo2d.cpp (hunk 1 above)
Comment (PeterPetrik) [reply mdal/frmts/mdal_flo2d.cpp:512]: it is ok

**Review by PeterPetrik — APPROVED** [review]

**Pull request merged**